tealish/nodes.py

```diff
@@ -1565,23 +1569,13 @@ def process(self) -> None:
         for n in self.nodes:
             n.process()
 
-        struct: Dict[str, Any] = {
-            "fields": {},
-            "size": 0,
-        }
-        offset = 0
-        for n in self.child_nodes:
-            # TODO: again child nodes are not the type
-            # we expect (BaseNode not StructFieldDef)
-            n = cast(StructFieldDefinition, n)
-            struct["fields"][n.field_name] = {
-                "type": n.data_type,
-                "size": n.size,
-                "offset": offset,
-            }
-            offset += n.size
-        struct["size"] = offset
-        self.define_struct(self.name, struct)
+        # TODO: unsafe cast?
+        define_struct(
+            self.name,
+            TealishStructDefinition(
+                cast(List[StructFieldDefinition], self.child_nodes)
+            ),
+        )
```

Comment:
I wonder do we need a new class for `TealishStructDefinition` or could we just have a registry of struct node objects? What do you see as the use for the separation between the node subclass and the definition class?

Reply:
that seems reasonable, will update